### DISCUSSION TITLE
Help relieving pain in testing (Storybook or Vitest)?

### DISCUSSION THREAD

**Discussion by charlieforward9** [vitest-dev/vitest#5881] [Q&A]
I am getting quite fatigued with testing...

I have set up our repo with Vitest, RTL and msw over the past 6-8 months using the JSDOM environment. I have had issues setting up almost every single test in my repo, spending days debugging components that work correctly in dev and fail in tests, making testing more of a burden than a strategy...

Given these discrepancies, I tried to get Browser mode set up with 1.6.0, and spent hours dealing with various bugs that made me revert back to our original (JSDOM) setup. Bumping to latest beta caused `peerDeps` problems...

I learned about Storybook and got that set up in a few hours with some simple core components. I like being able to visualize the components in isolation, but Storybook runs with Jest and Playwright. I like the fact that Vitest is designed specifically for Vite, and don't want to lose this with everything I have already set up around Vitest. However, I am in a hurricane of testing errors, especially with browser mode, while the dev server sails smoothly. 

Ultimately my question is:
* What can't I do in Storybook that I would need Vitest for?
* Is there a projected timeline for stable Browser mode?

I know these are vague questions but I am simply looking for some pointers around how to build the testing layer of our codebase with confidence and efficiency.

**Comment by charlieforward9** [comment]
Reading up on [this](https://github.com/vitest-dev/vitest/discussions/5828) now.

**Reply by jnarowski** [reply]
I am in the same boat. We're evaluating Vitest 2x VS Storybook for unit and component testing purposes. I find the visual rendering incredibly helpful when writing tests. Curious what you ended up doing? It seems like Vitest 2.0 might be able to handle it all. We have some apps that can run Vitest but not Storybook so I'm hoping to do it all with Vitest (everything but e2e)

**Reply by charlieforward9** [reply]
Seems like for now we are just using Vitest. We have storybook set up and I really enjoyed developing the component in the Storybook environment, but dont use it too much cause I just have the habit of testing the components within the development server environment.
